File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/consumer/InvoiceConsumer.scala
package com.example.events.consumer

import cats.effect.IO
import com.example.events.Invoice
import com.example.events.header.StandardHeaders
import java.lang.AutoCloseable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords
import cats.syntax.all._
import scala.jdk.CollectionConverters._

/** Type-safe consumer for invoice topic */
case class InvoiceConsumer(
  consumer: Consumer[String, Invoice],
  handler: InvoiceHandler,
  topic: String = "invoice"
) extends AutoCloseable {
  /** Poll for messages and dispatch to handler, returning composed effect */
  def poll(timeout: Duration): IO[Unit] = {
    val records: ConsumerRecords[String, Invoice] = consumer.poll(timeout)
    return records.asScala.toList.traverse_(record => handler.handle(record.key, record.value, StandardHeaders.fromHeaders(record.headers)))
  }

  /** Close the consumer */
  override def close: Unit = {
    consumer.close
  }
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/InvoiceConsumer.kt
package combined.avro_events.consumer

import com.example.events.Invoice
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Multi
import io.smallrye.mutiny.Uni
import java.io.Closeable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords

/** Type-safe consumer for invoice topic */
data class InvoiceConsumer(
  val consumer: Consumer<kotlin.String, Invoice>,
  val handler: InvoiceHandler,
  val topic: kotlin.String = "invoice"
) : Closeable {
  /** Close the consumer */
  override fun close() {
    consumer.close()
  }

  /** Poll for messages and dispatch to handler, returning composed effect */
  fun poll(timeout: Duration): Uni<Unit> {
    val records: ConsumerRecords<kotlin.String, Invoice> = consumer.poll(timeout)
    return Multi.createFrom().iterable(records).onItem().transformToUniAndConcatenate({ record -> handler.handle(record.key(), record.value(), StandardHeaders.fromHeaders(record.headers())) }).collect().last()
  }
}